pkg/kotsadmupstream/upstream.go

```diff
@@ -66,6 +67,22 @@ func DownloadUpdate(appID string, update types.Update, skipPreflights bool, skip
 		}
 
 		errMsg := finalError.Error()
+
+		var kotsApplication *kotsv1beta1.Application
```

Comment:
i suggest moving https://github.com/replicatedhq/kots/pull/2563/files#diff-14ec3523e1b869eb00191bbeb8af0572feeea19c9e82de9ca3669936c492b87fR70-R85 to line 98 right before creating the pending download app version since before that it's not an incompatibility error.